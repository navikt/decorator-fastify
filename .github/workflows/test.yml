name: test

on:
  pull_request:
    branches: 
      - main

env:
  IMAGE: ghcr.io/${{ github.repository }}:${{ github.sha }}
jobs:
  test:
    name: test
    runs-on: ubuntu-latest
    permissions:
      contents: read
      id-token: write
      packages: write
    steps:
      - uses: actions/checkout@v3
      - uses: oven-sh/setup-bun@v1
        with:
          bun-version: '1.0.1'
      - name: Install dependencies
        run: bun install --immutable
        env:
          NODE_AUTH_TOKEN: ${{ secrets.READER_TOKEN }}
      - name: Build app
        run: bun run build
      - name: Run tests
        run: bun test