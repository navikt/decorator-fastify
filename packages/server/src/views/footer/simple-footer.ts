import globalCls from "decorator-client/src/styles/global.module.css";
import cls from "decorator-client/src/styles/simple-footer.module.css";
import utilCls from "decorator-client/src/styles/utilities.module.css";
import html from "decorator-shared/html";
import { Features, Link } from "decorator-shared/types";
import i18n from "../../i18n";
import { ScreenshareButton } from "./screenshare-button";

export const SimpleFooter = ({
    links,
    features,
}: {
    links: Link[];
    features: Features;
}) => html`
    <footer class="${cls.simpleFooter}">
        <div class="${cls.simpleFooterContent} ${utilCls.contentContainer}">
            <div class="${cls.footerLinkList}">
                ${links.map(
                    ({ url, content }) => html`
                        <a
                            href="${url}"
                            class="${globalCls["navds-link"]} ${cls.footerLink}"
                            >${content}</a
                        >
                    `,
                )}
            </div>
            ${features["dekoratoren.skjermdeling"] &&
            ScreenshareButton(i18n("share_screen"))}
        </div>
    </footer>
`;
