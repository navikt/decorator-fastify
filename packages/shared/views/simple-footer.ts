import { Personvern } from 'decorator-shared/utils';
import { Texts } from '../texts';
import { html } from '../utils';

export function SimpleFooter({
  personvern,
  texts,
}: {
  texts: Texts;
  personvern: Personvern;
}) {
  return html`
    <footer class="simple-footer">
      <style>
        .simple-footer {
          border-top: 2px solid #0067c5;
          background-color: white;
          margin: 0 auto;
          padding: 1rem 3rem;
        }

        .simple-footer__content {
          display: flex;
          align-items: center;
          justify-content: space-between;
        }

        .simple-footer__link-list {
          display: flex;
          gap: 2rem;
        }

        .simple-footer__link {
          padding: 0.25rem 0;
          color: #0067c5;
          display: flex;
          gap: 0.25rem;
          align-items: center;
          text-decoration: underline;
          text-decoration-thickness: 0.0625em;
          text-underline-offset: 0.15em;
        }
      </style>
      <div class="simple-footer__content">
        <ul class="simple-footer__link-list">
          ${personvern.map(
            (link) => html`
              <li>
                <a class="simple-footer__link" href="${link.path}"
                  >${link.displayName}</a
                >
              </li>
            `,
          )}
        </ul>
        <a class="simple-footer__link" href="#">
          ${texts.share_screen}<svg
            width="1em"
            height="1em"
            viewBox="0 0 24 24"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
            focusable="false"
            role="img"
          >
            <path
              fill-rule="evenodd"
              clip-rule="evenodd"
              d="M2.25 4.5c0-.69.56-1.25 1.25-1.25h17c.69 0 1.25.56 1.25 1.25v11c0 .69-.56 1.25-1.25 1.25h-7.75v2.5H19a.75.75 0 0 1 0 1.5H6a.75.75 0 0 1 0-1.5h5.25v-2.5H3.5c-.69 0-1.25-.56-1.25-1.25v-11Zm1.5.25v10.5h16.5V4.75H3.75Z"
              fill="currentColor"
            ></path>
          </svg>
        </a>
      </div>
    </footer>
  `;
}
