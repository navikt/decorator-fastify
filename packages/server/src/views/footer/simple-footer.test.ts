import { expect, test } from 'bun:test';
import { SimpleFooter } from './simple-footer';

const links = [
  {
    content: 'Personvern og informasjonskapsler',
    url: '/no/nav-og-samfunn/om-nav/personvern-i-arbeids-og-velferdsetaten',
  },
  {
    content: 'Tilgjengelighet',
    url: '/tilgjengelighet',
  },
];

test('renders simple footer', async () => {
  expect(
    SimpleFooter({
      links,
      texts: {
        share_screen: 'Del skjerm med veileder',
      },
    })(),
  ).toMatchSnapshot();
});
