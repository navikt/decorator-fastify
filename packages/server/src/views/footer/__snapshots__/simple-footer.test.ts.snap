// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`renders simple footer 1`] = `
"
  <footer class="simpleFooter">
    <div class="simpleFooterContent contentContainer">
      <div class="footerLinkList">
        
    <a
      is="lenke-med-sporing"
      role=""
      href="/no/nav-og-samfunn/om-nav/personvern-i-arbeids-og-velferdsetaten"
      id=""
      tabindex=""
      lang=""
      data-analytics-event-args='{"category":"dekorator-footer","action":"kontakt//no/nav-og-samfunn/om-nav/personvern-i-arbeids-og-velferdsetaten","label":"/no/nav-og-samfunn/om-nav/personvern-i-arbeids-og-velferdsetaten"}'
      data-class-name-override="_footerLink_yutxt_19"
      data-container-class-name=""
      data-class-name=""
      data-extra-attrs='undefined'
      
      data-default-style="true"
    >
    Personvern og informasjonskapsler
    </a>
  
    <a
      is="lenke-med-sporing"
      role=""
      href="/tilgjengelighet"
      id=""
      tabindex=""
      lang=""
      data-analytics-event-args='{"category":"dekorator-footer","action":"kontakt//tilgjengelighet","label":"/tilgjengelighet"}'
      data-class-name-override="_footerLink_yutxt_19"
      data-container-class-name=""
      data-class-name=""
      data-extra-attrs='undefined'
      
      data-default-style="true"
    >
    Tilgjengelighet
    </a>
  
      </div>
      <button type="button" class="screenshareButton">
    Del skjerm med veileder
    <svg
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      focusable="false"
      role="img"
    >
      <path
        fill-rule="evenodd"
        clip-rule="evenodd"
        d="M2.25 4.5c0-.69.56-1.25 1.25-1.25h17c.69 0 1.25.56 1.25 1.25v11c0 .69-.56 1.25-1.25 1.25h-7.75v2.5H19a.75.75 0 0 1 0 1.5H6a.75.75 0 0 1 0-1.5h5.25v-2.5H3.5c-.69 0-1.25-.56-1.25-1.25v-11Zm1.5.25v10.5h16.5V4.75H3.75Z"
        fill="currentColor"
      ></path>
    </svg>
  </button>
    </div>
  </footer>
"
`;
